src/MPS.jl

```diff
@@ -92,6 +97,64 @@ function MPS(arrays::Vector{<:AbstractArray}; order=defaultorder(MPS))
     return MPS(ansatz, NonCanonical())
 end
 
+function MPS(form::MixedCanonical, arrays; order=defaultorder(MPS), check=true)
+    mps = MPS(arrays; form=NonCanonical(), order, check)
+    mps.form = form
+
+    # Check mixed canonical form
+    check && check_form(form, mps)
+
+    return mps
+end
+
+function MPS(::Canonical, arrays, λ; order=defaultorder(MPS), check=true)
+    @assert length(λ) == length(arrays) - 1 "Number of λ tensors must be one less than the number of arrays"
+    @assert all(==(1) ∘ ndims, λ) "All λ tensors must be Vectors"
+
+    mps = MPS(arrays; form=NonCanonical(), order, check)
+    mps.form = Canonical()
+
+    # Create tensors from 'λ'
+    map(enumerate(λ)) do (i, array)
+        tensor = Tensor(array, (inds(mps; at=Site(i), dir=:right),))
+        push!(mps, tensor)
+    end
+
+    # Check canonical form by contracting Γ and λ tensors and checking their orthogonality
+    check && check_form(Canonical(), mps)
+
+    return mps
+end
+
+function check_form(::MixedCanonical, mps::AbstractMPO)
+    orthog_center = form(mps).orthog_center
+    for i in 1:nsites(mps)
+        if i < id(orthog_center) # Check left-canonical tensors
+            isisometry(mps, Site(i); dir=:right) || throw(ArgumentError("Tensors are not left-canonical"))
+        elseif i > id(orthog_center) # Check right-canonical tensors
+            isisometry(mps, Site(i); dir=:left) || throw(ArgumentError("Tensors are not right-canonical"))
+        end
+    end
+
+    return true
+end
+
+function check_form(::Canonical, mps::AbstractMPO)
+    for i in 1:nsites(mps)
+        if i > 1
+            isisometry(contract(mps; between=(Site(i - 1), Site(i)), direction=:right), Site(i); dir=:right) ||
+                throw(ArgumentError("Can not form a left-canonical tensor in Site($i) from Γ and λ contraction."))
+        end
+
+        if i < nsites(mps)
+            isisometry(contract(mps; between=(Site(i), Site(i + 1)), direction=:left), Site(i); dir=:left) ||
+                throw(ArgumentError("Can not form a right-canonical tensor in Site($i) from Γ and λ contraction."))
```

Comment:
I think it's clearer if you fuse these 2 conditions:
```suggestion
        if i < nsites(mps) && isisometry(contract(mps; between=(Site(i), Site(i + 1)), direction=:left), Site(i); dir=:left)
            throw(ArgumentError("Can not form a right-canonical tensor in site $i from Γ and λ contraction."))
```